wai-aria/role/invalid-roles.html

```diff
@@ -8,29 +8,127 @@
+<span role=" " data-testname="span with empty character as role" class="ex-generic">x</span>
+<span role="&#13" data-testname="span with line break (&#13) character as role" class="ex-generic">x</span>
+<span role="&#9" data-testname="span with tab (&#9) as role (should be treated as whitespace)" class="ex-generic">x</span>
+<span role="&#8203" data-testname="span with zero-width space as role (should be treated as whitespace)" class="ex-generic">x</span>
+<span role="&#10240" data-testname="span with braille space (10240) as role" class="ex-generic">x</span>
+<span role="&#x2800" data-testname="span with braille space (x2800) as role" class="ex-generic">x</span>
+<span role="&nbsp" data-testname="span with non-breaking space (nbsp) as role" class="ex-generic">x</span>
+<span role="&#20" data-testname="span with standard space (nbsp) as role" class="ex-generic">x</span>
+
+<!-- Unescaped whitespace tests with <span> (including line breaks, tabs, zero-width space, braille space, non-breaking space, standard space) -->
+  <span role=" " data-testname="span with empty character as role" class="ex-generic">x</span>
+  <span role="
+  " data-testname="span with line break (&#13) character as role" class="ex-generic">x</span>
+  <span role="  " data-testname="span with tab (&#9) as role (should be treated as whitespace)" class="ex-generic">x</span>
+  <span role="‍" data-testname="span with zero-width space as role (should be treated as whitespace)" class="ex-generic">x</span>
+  <span role="⠀" data-testname="span with braille space (10240) as role" class="ex-generic">x</span>
+  <span role=" " data-testname="span with non-breaking space (nbsp) as role" class="ex-generic">x</span>
+  <span role=" " data-testname="span with standard space as role" class="ex-generic">x</span>
+
+<!-- Diacritics with <a> -->
+<a href="https://www.apple.com/" role="&#771" data-testname="link with role set to tilde diacritic" data-expectedrole="link" class="ex">x</a>
+<a href="https://www.apple.com/" role="foo&#771" data-testname="link with role foo that has tilde diacritic" data-expectedrole="link" class="ex">x</a>
+<a href="https://www.apple.com/" role="&#773" data-testname="link with role set to overline diacritic" data-expectedrole="link" class="ex">x</a>
+<a href="https://www.apple.com/" role="foo&#773" data-testname="link with role foo that has overline diacritic" data-expectedrole="link" class="ex">x</a>
+<a href="https://www.apple.com/" role="&#813" data-testname="link with role set to circumflex diacritic below" data-expectedrole="link" class="ex">x</a>
+<a href="https://www.apple.com/" role="foo&#813" data-testname="link with role foo that has circumflex diacritic below" data-expectedrole="link" class="ex">x</a>
+<a href="https://www.apple.com/" role="button&#771" data-testname="link with role set to button with tilde diacritic" data-expectedrole="link" class="ex">x</a>
```

Comment:
`href="#"` should be sufficient, but for future reference if you need a fully qualified domain, https://example.com/ is reserved for that.

```suggestion
<a href="#" role="&#771" data-testname="link with role set to tilde diacritic" data-expectedrole="link" class="ex">x</a>
<a href="#" role="foo&#771" data-testname="link with role foo that has tilde diacritic" data-expectedrole="link" class="ex">x</a>
<a href="#" role="&#773" data-testname="link with role set to overline diacritic" data-expectedrole="link" class="ex">x</a>
<a href="#" role="foo&#773" data-testname="link with role foo that has overline diacritic" data-expectedrole="link" class="ex">x</a>
<a href="#" role="&#813" data-testname="link with role set to circumflex diacritic below" data-expectedrole="link" class="ex">x</a>
<a href="#" role="foo&#813" data-testname="link with role foo that has circumflex diacritic below" data-expectedrole="link" class="ex">x</a>
<a href="#" role="button&#771" data-testname="link with role set to button with tilde diacritic" data-expectedrole="link" class="ex">x</a>
```

Reply:
TIL https://example.org is also reserved.